### PR TITLE
fix: update revoke ui with error handling

### PR DESCRIPTION
**Pull request by isaackps**
## Summary

Update revoke UI with error handling and better error messages

## Changes

- added error handling same as TT web
- updated the UI based on feedbacks
- fix some typos

## Issues

https://www.pivotaltracker.com/n/projects/2424361/stories/179279240

### CODE REVIEW

**Comment by cavacado** [comment]
dont put 2 places with the same logic
lets think of a better place to put it

File: src/components/RevokeContainer/RevokeContainer.tsx [hunk 1]
```diff
@@ -25,19 +26,24 @@ export const RevokeContainer: FunctionComponent = () => {
...
           const verify = verificationBuilder(openAttestationVerifiers, { network: network });
           const fragments = await verify(revokeDocuments[0]);
-          isDocumentValid = isValid(fragments);
+          errors = errorMessageHandling(fragments);
         }
 
-        if (!isDocumentValid) {
-          setErrorMessage("Document cannot be read. Please check that you have a valid document");
+        if (errors.length > 0) {
+          setErrorMessage(CONSTANTS.MESSAGES[errors[0]].failureMessage);
```
Comment (simboonlong) [inline src/components/RevokeContainer/RevokeContainer.tsx:39]: why only `[0]` ? show all pushed errors?

File: src/components/RevokeContainer/RevokeContainer.tsx [hunk 2]
```diff
@@ -25,19 +26,24 @@ export const RevokeContainer: FunctionComponent = () => {
...
           setRevokeDocuments([]);
           setDocumentUploadState(DocumentUploadState.ERROR);
         } else {
+          isDocumentValid = true;
+        }
+
+        if (isDocumentValid) {
```
Comment (simboonlong) [inline src/components/RevokeContainer/RevokeContainer.tsx:46]:
```suggestion

```
rm?

File: src/components/UI/StyledDropZone/StyledDropZone.tsx [hunk 3]
```diff
@@ -75,24 +83,34 @@ export const StyledDropZone: FunctionComponent<DropZoneProps> = ({
 
       {error && <p className="max-w-lg text-rose text-lg leading-none font-bold mb-2">Error</p>}
       {fileTypeError && (
-        <p className="max-w-lg text-rose text-lg leading-none mb-2" data-testid="invalid-file-error">
-          Incorrect file type selected, Only {dropzoneOptions.accept} are allowed
-        </p>
+        <ErrorMessage
+          id="invalid-file-error"
+          message={`Incorrect file type selected, Only ${dropzoneOptions.accept} are allowed`}
```
Comment (simboonlong) [inline src/components/UI/StyledDropZone/StyledDropZone.tsx:88]: we could start consolidating these dropzone error codes + custom error messages somewhere, wdyt?

Comment (isaackps) [reply src/components/UI/StyledDropZone/StyledDropZone.tsx:88]: yeah we can, most of them regarding the validation of documents are in the tt-utils repo. we can create another story to consolidate them.

File: src/components/RevokeContainer/RevokeContainer.tsx (hunk 2 above)
Comment (isaackps) [reply src/components/RevokeContainer/RevokeContainer.tsx:46]: ok

File: src/components/RevokeContainer/RevokeContainer.tsx (hunk 1 above)
Comment (isaackps) [reply src/components/RevokeContainer/RevokeContainer.tsx]: ok updated it to show all errors

File: src/components/RevokeContainer/RevokeDocumentDropZone/RevokeDocumentDropZone.tsx [hunk 4]
```diff
@@ -30,14 +32,15 @@ export const RevokeDocumentDropZone: FunctionComponent<RevokeDocumentDropZone> =
 
   useEffect(() => {
     if (documentUploadState === DocumentUploadState.ERROR) {
-      const readFileError = new Error(
-        errorMessage ?? "Document cannot be read. Please check that you have a valid document"
-      );
-      setFileErrors([readFileError]);
+      const readFileError =
+        errorMessages !== undefined && errorMessages.length > 0
```
Comment (simboonlong) [inline src/components/RevokeContainer/RevokeDocumentDropZone/RevokeDocumentDropZone.tsx:36]: how about setting `errorMessages` prop as required, so no need to check for undefined here?

File: src/components/UI/StyledDropZone/StyledDropZone.tsx [hunk 5]
```diff
@@ -75,24 +83,34 @@ export const StyledDropZone: FunctionComponent<DropZoneProps> = ({
...
       )}
       {fileErrors &&
         fileErrors.length > 0 &&
         fileErrors.map((formError, index: number) => {
           return (
-            <p key={index} className="max-w-lg text-rose text-lg leading-none mb-2" data-testid="file-error">
-              {formError instanceof Error ? "" : formError.instancePath} {formError.message}
-            </p>
+            <ErrorMessage
+              key={index}
```
Comment (simboonlong) [inline src/components/UI/StyledDropZone/StyledDropZone.tsx:110]:
```suggestion
              key={`file-errors-${index}`}
```

File: src/components/RevokeContainer/RevokeDocumentDropZone/RevokeDocumentDropZone.tsx (hunk 4 above)
Comment (isaackps) [reply src/components/RevokeContainer/RevokeDocumentDropZone/RevokeDocumentDropZone.tsx:36]: ok sure

File: src/components/UI/StyledDropZone/StyledDropZone.tsx (hunk 5 above)
Comment (isaackps) [reply src/components/UI/StyledDropZone/StyledDropZone.tsx:110]: any reason for this change to the key attribute? it wont be display anywhere right?

Comment (simboonlong) [reply src/components/UI/StyledDropZone/StyledDropZone.tsx:110]: last resort -> https://reactjs.org/docs/lists-and-keys.html#keys, ok to ignore also

Comment (isaackps) [reply src/components/UI/StyledDropZone/StyledDropZone.tsx:110]: ok updated

**Review by simboonlong — APPROVED** [review]
lgtm

**Pull request merged**